### PR TITLE
pagination + nonfunctional filter

### PR DESCRIPTION
**Pull request by emilyliublair**
### Description

Closes (https://github.com/GTBitsOfGood/focus-ga/issues/18)

### Type of change

🤕 Bug Fix
🆕 Feature
🔁 Chore/Cleanup
✨ Refactor

### Checklist

- [ ]  Code follows design and style guidelines
- [ ]  Code is commented with doc blocks
- [ ]  Latest code has been rebased from base branch (usually `develop`)
- [ ]  Commits follow guidelines (concise, squashed, etc)
- [ ]  Github issues have been linked in relevant commits
- [ ]  Relevant reviewers (EM) have been assigned to this PR

### CODE REVIEW

File: src/app/(community)/page.tsx [hunk 1]
```diff
@@ -1,17 +1,139 @@
...
 export const dynamic = 'force-dynamic';
 
-export default async function Home() {
-  const posts = await getPopulatedPosts();
+export default function Home() {
+  const [posts, setPosts] = useState<PopulatedPost[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [hasMore, setHasMore] = useState(true);
+  const [page, setPage] = useState(0);
+  const limit = 5;
```
Comment (nathangong) [inline src/app/(community)/page.tsx:20]: create this as a PAGINATION_LIMIT constant in `consts.ts`. set it to 10 for now?

File: src/app/(community)/page.tsx [hunk 2]
```diff
@@ -1,17 +1,139 @@
...
+                  <PostComponent post={post} clickable={true} />
+                </div>
+              );
+            } else {
+              return <PostComponent key={post._id} post={post} clickable={true} />;
+            }
+          })}
+          {loading && 
+            <div className="flex items-center justify-center">
+              <LoaderCircle className="animate-spin text-blue-500" size={32}/>
```
Comment (nathangong) [inline src/app/(community)/page.tsx:133]: the color doesn't seem to be working for the loading indicator (might just be on my computer tho?), lmk if its the same on yours too

File: src/components/FilterComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+type FilterProps = {
+  filters: Filter<any>[];
+}
+
+export default function FilterComponent( props: FilterProps ) {
+  return (
+    <div className="relative inline-block w-full flex space-x-4 items-center">
+      <label className="block text-sm font-medium">Filter By:</label>
+        {props.filters.map((filter, index) => (
+          <DropdownComponent key = {index} filter={filter} />
```
Comment (nathangong) [inline src/components/FilterComponent.tsx:15]:
```suggestion
          <DropdownComponent key={index} filter={filter} />
```

File: src/components/DropdownComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { Filter } from "@/utils/consts"
+
+type DropdownProps = {
+  filter: Filter<any>;
+};
+
+export default function DropdownComponent<T extends { _id: string; name: string }>(
```
Comment (nathangong) [inline src/components/DropdownComponent.tsx:16]: what is this type parameterization enforcing?

**Review by nathangong — CHANGES_REQUESTED** [review]
functionality looks great overall! just some small nitpicks here and there

File: src/app/(community)/page.tsx (hunk 2 above)
Comment (emilyliublair) [reply src/app/(community)/page.tsx:133]: oops didn't mean to have the color anyways

**Review by nathangong — APPROVED** [review]

**Pull request merged**